tests/plugins/test_amp_plugins.py

```diff
@@ -69,18 +70,34 @@ def test_amp_apex_ddp(
     assert isinstance(trainer.precision_plugin, plugin_cls)
 
 
+class CheckOnBeforeBackward(Callback):
```

Comment:
what if instead of adding these callbacks per plugin test you parameterize the test in `test_hooks`?

Reply:
Hey @carmocca,
I prefer the current approach as testing `on_before_backward` for all plugins will result in a large increase of CI time.
I think it is light enough to be embedded into existing tests.

Reply:
Note that we also get the advantage of checking all other hooks, not just `on_before_backward`.

The hook order should be consistent among accelerators/plugins/training_types so we eventually want to test it